### ISSUE TITLE
Idea: Filter hangar and buyback on all tags

### ISSUE THREAD

**Issue by cintaracorp** [waibcam/RSI_Companion#4]
Adding the ability to search on tags or buttons to require/exclude any of the tags would be nice. For instance, see only loaners, or not see any upgrades.

A simple UI would be to list all the tags that the view can present, click once to require that ships have the tag (boldface), click again to exclude the tag (overstrike), and click again to ignore (normal presentation).

**Comment by waibcam** [comment]
All the tags are already listed in the dropdown filter in "My ships" (Manufacturers, Status, Focus and Type).
Then, you need to type "owned" or "loaner" in the search filter to see them.

If you have, for instance, multiple Avanger, you can search for "owned avenger" and you'll see all of them.

I realize it's not what you proposed but it sounds enough in my humble opinion, no?

**Comment by cintaracorp** [comment]
Yes, the search field works, and you can filter for everything there of course (though I haven't found boolean operators, so I can't negate a search). The reason I posted was for buy-backs. I wanted to filter out all my upgrades, but failed. I suppose a drop-down like the rest for "standalone", "upgrade", "package" would be enough.

It also seems that I don't see my upgrades in my hangar, only in my buy-backs. Is that right?

**Comment by waibcam** [comment]
> The reason I posted was for buy-backs. I wanted to filter out all my upgrades, but failed. I suppose a drop-down like the rest for "standalone", "upgrade", "package" would be enough. LTI is also missing, if I do a search for LTI it filters out far too much.

That works for me:
Upgrade: https://i.imgur.com/Y7h9dPF.jpg
LTI: https://i.imgur.com/Qxj6U39.jpg

However, I'll check to add a dropdown list, as that might be easier to use.

> It also seems that I don't see my upgrades in my hangar, only in my buy-backs. Is that right?

Yep you can't see stuff that you haven't applied basically. It's a choice by design.
